[lldb] Change breakpoint interfaces for error handling

JDevlieghere opened the pull request:
This RP changes some Breakpoint-related interfaces to return errors. On its own these improvements are small, but they encourage better error handling going forward. There are a bunch of other candidates, but these were the functions that I touched while working on #146602.

jasonmolenda reviewed (approved):
Seems like a reasonable RP to me.

jimingham reviewed (approved):
LGTM too.

JDevlieghere committed:
[lldb] Change breakpoint interfaces for error handling

This RP changes some Breakpoint-related interfaces to return errors. On
its own these improvements are small, but they encourage better error
handling going forward. There are a bunch of other candidates, but these
were the functions that I touched while working on #146602.

JDevlieghere committed:
Formatting
